### ISSUE TITLE
Could not start workspace workspace-man. Reason: Start of environment 'workspace-man' failed. Error: Can't create folder for the logs of machine

### ISSUE THREAD

**Issue by ghost** [eclipse-che/che#3929]
I was advised in #3923 to write my own Dockerfile to workaround the stack ui issue described in #3883 . However, doing that I cannot launch the resulting workspace, and I get this error:
```
Could not start workspace workspace-man. Reason: Start of environment 'workspace-man' failed. Error: Can't create folder for the logs of machine
```

**Reproduction Steps:**
1. Create new Workspace + Project in one go at the initial dashboard view you get on first launch with no workspaces/projects present, and choose "Stack Authoring" with a custom Dockerfile:
```
FROM Ubuntu

RUN apt-get install -y git python3 gcc gcc-c++
```
2. Wait for loading screen to show up. It will not terminate, so navigate away to the actual workspace listed on the left
3. It will show the IDE interface in an attempt to open up the workspace, and then show the error: `Could not start workspace workspace-man. Reason: Start of environment 'workspace-man' failed. Error: Can't create folder for the logs of machine`

**Che version:**
```
Your CLI version is 'nightly'.
Your installed version is '<not-installed>'.
```
(don't ask me why, it runs fine and didn't show `<not-installed>` before)

**OS and version:**    Fedora 25 / Linux cyberman 4.9.3-200.fc25.x86_64 #1 SMP Fri Jan 13 01:01:13 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Docker version:**
```
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:58:06 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:58:06 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Che cli.log output:**
```
[root@cyberman eclipse-che]# cat data/cli.log 
Fri Jan 27 09:40:59 UTC 2017
INFO: (che cli): Checking network... (hint: '--fast' skips nightly, version, network, and preflight checks)
/usr/bin/curl  "-I -k dockerhub.com -s -o /dev/null --write-out %{http_code}"
Checking registry for version 'nightly' images
eval BOOTSTRAP_IMAGE_ALPINE=alpine:3.4
eval BOOTSTRAP_IMAGE_CHEIP=eclipse/che-ip:nightly
eval IMAGE_INIT=eclipse/che-init:nightly
eval IMAGE_CHE=eclipse/che-server:nightly
eval UTILITY_IMAGE_CHEACTION=eclipse/che-action:nightly
eval UTILITY_IMAGE_CHEDIR=eclipse/che-dir:nightly
eval UTILITY_IMAGE_CHETEST=eclipse/che-test:nightly
eval UTILITY_IMAGE_CHEMOUNT=eclipse/che-mount:nightly
INFO: (che cli):  Loading cli...
DEBUG: docker_run
DEBUG: update_image
INFO: (che download):  Pulling image eclipse/che:nightly

docker pull eclipse/che:nightly >> "/data/cli.log" 2>&1

DEBUG: is_initialized
DEBUG: cli_parse
DEBUG: cmd_start
DEBUG: is_initialized
DEBUG: is_docker_for_mac
docker inspect -f '{{.Id}}' che
/usr/bin/curl  "-I -k 172.17.0.1:8080/api/ -s -o /dev/null --write-out %{http_code}"
200
INFO: (che start):  Eclipse Che is already running
INFO: (che start):  Server logs at "docker logs -f che"
DEBUG: is_initialized
INFO: (che start):  Ver: <not-configed>
INFO: (che start):  Use: http://172.17.0.1:8080
INFO: (che start):  API: http://172.17.0.1:8080/swagger
Fri Jan 27 12:14:22 UTC 2017
INFO: (che cli): Checking network... (hint: '--fast' skips nightly, version, network, and preflight checks)
/usr/bin/curl  "-I -k dockerhub.com -s -o /dev/null --write-out %{http_code}"
Checking registry for version 'nightly' images
eval BOOTSTRAP_IMAGE_ALPINE=alpine:3.4
eval BOOTSTRAP_IMAGE_CHEIP=eclipse/che-ip:nightly
eval IMAGE_INIT=eclipse/che-init:nightly
eval IMAGE_CHE=eclipse/che-server:nightly
eval UTILITY_IMAGE_CHEACTION=eclipse/che-action:nightly
eval UTILITY_IMAGE_CHEDIR=eclipse/che-dir:nightly
eval UTILITY_IMAGE_CHETEST=eclipse/che-test:nightly
eval UTILITY_IMAGE_CHEMOUNT=eclipse/che-mount:nightly
INFO: (che cli):  Loading cli...
DEBUG: docker_run
DEBUG: update_image
INFO: (che download):  Pulling image eclipse/che:nightly

docker pull eclipse/che:nightly >> "/data/cli.log" 2>&1

DEBUG: is_initialized
DEBUG: cli_parse
DEBUG: cmd_version
Your CLI version is 'nightly'.
DEBUG: is_initialized
Your installed version is '<not-installed>'.

Available on DockerHub:
/usr/bin/curl  "-s https://hub.docker.com/v2/repositories/eclipse/che/tags/"
  nightly
  latest
  5.1.2
  5.1.1
  5.1.0
  5.0.1
  5.0.0
  5.0.0-latest
  5.0.0-M8
  5.0.0-M7
  See older versions at: "https://hub.docker.com/v2/repositories/eclipse/che/tags/?page=2"
Fri Jan 27 12:15:14 UTC 2017
INFO: (che cli): Checking network... (hint: '--fast' skips nightly, version, network, and preflight checks)
/usr/bin/curl  "-I -k dockerhub.com -s -o /dev/null --write-out %{http_code}"
Checking registry for version 'nightly' images
eval BOOTSTRAP_IMAGE_ALPINE=alpine:3.4
eval BOOTSTRAP_IMAGE_CHEIP=eclipse/che-ip:nightly
eval IMAGE_INIT=eclipse/che-init:nightly
eval IMAGE_CHE=eclipse/che-server:nightly
eval UTILITY_IMAGE_CHEACTION=eclipse/che-action:nightly
eval UTILITY_IMAGE_CHEDIR=eclipse/che-dir:nightly
eval UTILITY_IMAGE_CHETEST=eclipse/che-test:nightly
eval UTILITY_IMAGE_CHEMOUNT=eclipse/che-mount:nightly
INFO: (che cli):  Loading cli...
DEBUG: docker_run
DEBUG: update_image
INFO: (che download):  Pulling image eclipse/che:nightly

docker pull eclipse/che:nightly >> "/data/cli.log" 2>&1

DEBUG: is_initialized
DEBUG: cli_parse
DEBUG: cmd_version
Your CLI version is 'nightly'.
DEBUG: is_initialized
Your installed version is '<not-installed>'.

Available on DockerHub:
/usr/bin/curl  "-s https://hub.docker.com/v2/repositories/eclipse/che/tags/"
  nightly
  latest
  5.1.2
  5.1.1
  5.1.0
  5.0.1
  5.0.0
  5.0.0-latest
  5.0.0-M8
  5.0.0-M7
  See older versions at: "https://hub.docker.com/v2/repositories/eclipse/che/tags/?page=2"
[root@cyberman eclipse-che]#
```

**Comment by TylerJewell** [comment]
Your configuration looks good. I do not think this particular issue is a bug in our system, but may be related to how the Dockerfile is authored. Dockerfiles build slightly differently in the cloud than they do on your host system, so certain quirks apply.  We usuallly detect and warn for each of these, but there are many types of Dockerfiles, so we maybe missing certain edge cases.

@bmicklea - can we please have this in the support engineer backlog for investigation and assistance?

**Comment by bmicklea** [comment]
@eivantsov please ask one of the engineers on support rotation to investigate.

**Comment by ghost** [comment]
@JonasT can you check if SElinux is on in your Fedora. This issue seems to have nothing to do with the Dockerfile. It's a workspace sever trying to create a directory for machine logs and fails to do so. It should happen to all of your workspaces in fact. Or ready to go stacks are OK?

**Comment by ghost** [comment]
@eivantsov without SELinux being disabled, eclipse che doesn't even start properly for me. So yes, it is in permissive mode / off:

```
jonas@cyberman:~$ sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive
Mode from config file:          error (Permission denied)
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      30
jonas@cyberman:~$
```
You are correct about this being apparently unrelated to the Dockerfile:

I just tried "Create Workspace", then in "Runtime" picked the default "Java" one, then I clicked "Save" in the top right and "Run" -> `Start of environment 'default' failed. Error: Can't create folder for the logs of machine` (same error)

**Comment by ghost** [comment]
@JonasT I have tried it on a fresh Fedora on Digital Ocean. Ran just fine. 

By the way, your Dockerfile cannot be built locally - some issues with Ubuntu 16 and packages you are try to install.

As to `cannot create logs error` you should probably inspect system logs for clues.

It is strange since `/instance` directory got created, right? And you can see a bunch of sub dirs there.

Can you try the following?

```
[root@fedora-4gb-fra1-01 logs]# docker exec -ti che bash
bash-4.3# cd logs/
bash-4.3# ls -la
total 16
drwxr-xr-x    4 root     root          4096 Jan 28 20:05 .
drwxr-xr-x    1 root     root          4096 Jan 28 20:05 ..
drwxr-xr-x    3 root     root          4096 Jan 28 20:05 2017
drwxr-xr-x    7 root     root          4096 Jan 28 20:12 machines
bash-4.3# cd machines/
bash-4.3# mkdir myMachine
```
Logs directory is mounted into your Fedora's `/instance/logs dir`. `docker inspect che` will give you this:

```
    "Volumes": {
        "/conf": "/root/instance/config",
        "/data": "/root/instance/data",
        "/logs": "/root/instance/logs",
        "/var/run/docker.sock": "/var/run/docker.sock"
}
```

Try to play with creating files manually both in a container and locally.

**Comment by ghost** [comment]
Actually, while inspect lists those paths:
```
        "HostConfig": {
            "Binds": [
                "/home/jonas/Develop/eclipse-che/data/instance/config:/conf:rw",
                "/home/jonas/Develop/eclipse-che/data/instance/data:/data:rw",
                "/var/run/docker.sock:/var/run/docker.sock:rw",
                "/home/jonas/Develop/eclipse-che/data/instance/logs:/logs:rw"
            ],
```
.. inside the actual host directory, there is no "instance" folder:
```
[root@cyberman jonas]# ls /home/jonas/Develop/eclipse-che/data
cli.log
[root@cyberman jonas]#
```
Edit: Also, in the volumes section the volumes aren't actually properly set as I just noticed now:
```
            "Volumes": {
                "/conf": {},
                "/data": {},
                "/logs": {},
                "/var/run/docker.sock": {}
            },
```
Is this possibly related to #3863 with some other part of eclipse che still failing to deal with docker-compose-provided volumes on Fedora?

**Comment by ghost** [comment]
@JonasT can you start Che with the following volume mount?

`-v /home/jonas/Develop/eclipse-che/data:/data:Z`

Let's see if `:Z` suffix helps. The fact is that if a directory does not exist, Docker creates it for you when volume mount happens. Of course, only in case it has the power to do so.

**Comment by ghost** [comment]
It seems with the fix of #3863 and possibly other changes going into master 1-2 days ago, me mounting a custom-fixed `docker.sh` into eclipse-che (with just the tiny `*$1` -> `*$1*` change) actually made something break, causing this issue.

I removed that now, and workspace creation is fixed :smile:

Sorry about me ultimately causing it myself, I just had no other way to launch before hence I was still having that additional hack in use.
